src/firebolt/db/connection.py

```diff
@@ -42,18 +42,18 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         # Holding this lock for write means that connection is closing itself.
         # cursor() should hold this lock for read to read/write state
         self._closing_lock = RWLockWrite()
+        self._async_job_thread = AsyncJobThread()
 
-    @wraps(AsyncBaseConnection.cursor)
     def cursor(self) -> Cursor:
         with self._closing_lock.gen_rlock():
-            c = super().cursor()
+            c = super()._cursor(async_job_thread=self._async_job_thread)
```

Comment:
Would this work with multiple cursors from a single connection? And if multiple cursors are not supported can we lock it down somehow?

Reply:
Yes, the point is that every cursor need to share the same async_job_thread

Reply:
Ok, got it.